test/common-algorithms/dynamic-programming/sequential/coin_change.chpl

```diff
@@ -0,0 +1,31 @@
+//Time Complexity: O(mn)
+proc count(S, m:int, n:int):int{
+	// table[i] will be storing the number of solutions for
+	// value i. We need n+1 rows as the table is consturcted
+	// in bottom up manner using the base case (n = 0)
+	var table: [1..n+1] int = 0;
+	
+	// Initialize all table values as 0
+	table = 0;
+ 
+	// Base case (If given value is 0)
+	table(1) = 1;
+ 	
+	// Pick all coins one by one and update the table[] values
+	// after the index greater than or equal to the value of the
+	// picked coin
+	var i:int;
+	for i in (1..m){
+		var j:int;
+		for j in (S(i)..n){
+			table(j) += table(j-S(i)+1);
+ 		}
+	}
```

Comment:
Since the inner loop range depends on the outer loop, I think zip wouldn't work. I tried it and as expected, the following error occurs.
error: zippered iterations have non-equal lengths

Also, for the loop suggested above, when I try to compiler, it requires i and j to be declared. I am thinking that maybe this is a bug. What do you think?

Nevertheless, I have written a nested loop and removed looping variable declarations.